### PR TITLE
Tests for Continue as New and one additional scenario

### PR DESCRIPTION
**Pull request by agautam478**
**What changed?**
Cherry picked the two additional replay test scenarios from @taylanisikdemir PR https://github.com/uber-go/cadence-client/pull/1281

ff4f4d194181f9574c66336673ae33227ce30d96
ec3ecd5288eb3f9148bbce6910cf3c6173ce7115

**Why?**
We have decided not to move forward with the core changes in the PR as it requires additional extensive testing but these tests are extremely important and should be included in the replaytest suite. The `continue-as-new` test is particularly important as it will help us address one of the major bugs in the replayer.

**How did you test it?**
tested locally by running replay_test file

**Potential risks**
NA

### CODE REVIEW

File: test/replaytests/replay_test.go [hunk 1]
```diff
@@ -151,6 +151,17 @@ func TestBranchWorkflowWithExtraBranch(t *testing.T) {
 	assert.ErrorContains(t, err, "nondeterministic workflow")
 }
 
+// ####### my test
```
Comment (taylanisikdemir) [inline test/replaytests/replay_test.go:154]: my test? :) let's put a more descriptive comment

File: test/replaytests/sequantial_steps_workflow.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"go.uber.org/cadence/workflow"
+)
+
+/**
+ * This sample workflow executes sample activity 3 times sequentially.
+ */
+
+// sampleBranchWorkflow workflow decider
+func sequantialStepsWorkflow(ctx workflow.Context) error {
+	// starts activities in parallel
```
Comment (taylanisikdemir) [inline test/replaytests/sequantial_steps_workflow.go:36]: doesn't start activities in parallel

**Review by taylanisikdemir — APPROVED** [review]

File: test/replaytests/replay_test.go (hunk 1 above)
Comment (agautam478) [reply test/replaytests/replay_test.go:154]: will do. I cherry picked these test as they are.

File: test/replaytests/sequantial_steps_workflow.go (hunk 2 above)
Comment (agautam478) [reply test/replaytests/sequantial_steps_workflow.go]: Updated the test with the latest code.

**Pull request merged**